sepal_ui/scripts/decorator.py

```diff
@@ -60,27 +60,33 @@ def init_ee() -> None:
 
 
 @versionadded(version="3.0", reason="moved from utils to a dedicated module")
-def catch_errors(alert: v.Alert, debug: bool = False) -> Any:
+def catch_errors(alert: Optional[v.Alert] = None, debug: bool = False) -> Any:
     """Decorator to execute try/except sentence and catch errors in the alert message.
 
     If debug is True then the error is raised anyway.
 
     Args:
-        alert (sw.Alert): Alert to display errors
-        debug (bool): Whether to raise the error or not, default to false
+        alert: Alert to display errors
+        debug: Whether to raise the error or not, default to false
 
     Returns:
         The return statement of the decorated method
     """
 
     def decorator_alert_error(func):
         @wraps(func)
-        def wrapper_alert_error(*args, **kwargs):
+        def wrapper_alert_error(self, *args, **kwargs):
+
+            # Change name of variable to assign it again in this scope
+            alert_ = self.alert if not alert else alert
```

Comment:
what about if parent object has not defined an `alert` component, can we add an exception here?

Reply:
isn't it already the case ? if you don't set up a alert and the object don't have an alert object, you should get something like "xxx object hase no "alert" member"

Reply:
yes and not, you'll get an error, but that error will be too vague, that would be something like... "`None type doesn't have reset method`".
You don't really know where the alert is required, and perhaps you'll decide to create either a custom `w_alert` or `alert` and add it as parameter to the decorator, I mean, this feature would be in someway hidden, you'll have to really know how this has been written... or perhaps document it?...

Reply:
I think solved in https://github.com/12rambau/sepal_ui/pull/856/commits/95770ae6be052700d161fab194548f7fa541d467